src/server/connection.rs

```diff
@@ -1504,16 +1549,18 @@ impl Connection {
                 }
             });
             #[cfg(all(windows, feature = "flutter"))]
-            std::thread::spawn(|| {
+            std::thread::spawn(move || {
                 if crate::is_server() && !crate::check_process("--tray", false) {
-                    crate::platform::run_as_user(vec!["--tray"]).ok();
+                    crate::platform::run_as_user(vec!["--tray"], usid).ok();
                 }
             });
         }
     }
 
     async fn on_message(&mut self, msg: Message) -> bool {
         if let Some(message::Union::LoginRequest(lr)) = msg.union {
+            #[cfg(all(feature = "flutter", target_os = "windows"))]
+            self.handle_multiple_user_sessions(lr.clone()).await;
```

Comment:
Where do you tell this process run as system? You have said this can only work if it is installed and run as system service.